From 58b2e9d3d8c7a428ca2a9f503b0f47a2e898bf85 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Mon, 06 Apr 2020 08:10:17 +0300
Subject: [PATCH] [SW-10909] goya2: pll changes to enable legacy boot

This patch derives clocks for CPU, UART and SPI for goya2
using PCI PLL as its source.

Change-Id: I98326fd7d42559ac80a180680011af23be51b0d2
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/goya2/asic.c b/goya2/asic.c
index 99c770b..5c904e8 100644
--- a/goya2/asic.c
+++ b/goya2/asic.c
@@ -21,24 +21,27 @@
 
 u64 hl_pll_get_cpu_cycles_per_usec(void)
 {
-	u32 cpu_div0_freq = hl_pll_get_freq(mmPSOC_CPU_PLL_BASE, HL_PLL_DIV0);
+	/* As per Goya2 PLL cfg.xslx, PCI PLL Div[1] drives CPU */
+	u32 cpu_div0_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV1);
 
 	return cpu_div0_freq / USEC_PER_SEC;
 }
 
 void hl_uart_init(void)
 {
-	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
+	/* As per Goya2 PLL cfg.xlsx, PCI PLL Div[3] drives uart */
+	u32 uart_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV3);
 
-	preboot_uart_init(pcie_div2_freq / HL_UART_BAUDRATE);
+	preboot_uart_init(uart_clk_freq / HL_UART_BAUDRATE);
 }
 
 u32 hl_spi_baudr_get(void)
 {
 	u32 factor;
-	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
+	/* As per Goya2 PLL cfg.xlsx, PCI PLL Div[3] drives SPI */
+	u32 spi_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV3);
 
-	factor = pcie_div2_freq / HL_SPI_FLASH_MAX_FREQ;
+	factor = spi_clk_freq / HL_SPI_FLASH_MAX_FREQ;
 
 	return factor > 2 ? factor : 2;
 }
diff --git a/goya2/common.h b/goya2/common.h
index 3a65bd5..06d79c5 100644
--- a/goya2/common.h
+++ b/goya2/common.h
@@ -6,6 +6,7 @@
  */
 
 #include <preboot.h>
+#include <goya2_blocks.h>
 
 #define HL_SPI_NEXT_IMAGE_OFF		0x80000
 #define HL_SPI_MODIFI_IMAGE_OFF		0x80004
@@ -15,19 +16,4 @@
 #define PPBOOT_TEXT_BASE		0x7FF0000000
 #define PREBOOT_TEXT_BASE		(PPBOOT_TEXT_BASE + PPBOOT_SIZE)
 
-#define mmDMA_W_PLL_BASE		0x487000
-#define mmIF_W_PLL_BASE			0x488000
-#define mmIF_E_PLL_BASE			0x4A7000
-#define mmDMA_E_PLL_BASE		0x4A8000
-#define mmMESH_W_PLL_BASE		0x4C7000
-#define mmSRAM_W_PLL_BASE		0x4C8000
-#define mmMESH_E_PLL_BASE		0x4E7000
-#define mmSRAM_E_PLL_BASE		0x4E8000
-#define mmPSOC_CPU_PLL_BASE		0xC70000
-#define mmPSOC_MME_PLL_BASE		0xC71000
-#define mmPSOC_PCI_PLL_BASE		0xC72000
-#define mmPSOC_TPC_PLL_BASE		0xC73000
-#define mmNIC0_PLL_BASE			0xCF9000
-#define mmNIC1_PLL_BASE			0xD39000
-#define mmNIC2_MME_PLL_BASE		0xD79000
-#define mmNIC2_TPC_PLL_BASE		0xD7A000
+#define mmPSOC_PCI_PLL_BASE		(mmPSOC_PCI_PLL_CTRL_BASE & 0x00FFFFFF)
